java-util/src/main/java/io/druid/java/util/common/guava/ExecuteWhenDoneYielder.java

```diff
@@ -56,9 +56,10 @@ public boolean isDone()
   @Override
   public void close() throws IOException
   {
-    if (isDone()) {
+    boolean done = isDone();
+    baseYielder.close();
```

Comment:
there's also a problem with caching whereby serializing for cache is very expensive, but if you allow the cache to be calculated / populated in the background, then things like HLL buffers get recycled too soon. Not sure if such a thing is solvable at this point, but it would be nice if background caching worked right with how query resources are freed.